From 3d44d2fa86f0176cb2a565a45ff68bee34244134 Mon Sep 17 00:00:00 2001
From: Jishnu Prakash <quic_jprakash@quicinc.com>
Date: Thu, 23 May 2024 20:08:23 +0530
Subject: [PATCH 5/6] PENDING: dt-bindings: iio: Add ADC5 GEN3 Channel macros

Add macro definitions for ADC channels and virtual channels
per PMIC, to be used by clients of ADC5 Gen3 device.

Signed-off-by: Jishnu Prakash <quic_jprakash@quicinc.com>
Signed-off-by: Umang Chheda <quic_uchheda@quicinc.com>
Upstream-Status: Pending
---
 .../iio/qcom,spmi-adc5-gen3-pm8550.h          |  48 ++++++
 .../iio/qcom,spmi-adc5-gen3-pm8550b.h         |  97 ++++++++++++
 .../iio/qcom,spmi-adc5-gen3-pm8550vx.h        |  15 ++
 .../iio/qcom,spmi-adc5-gen3-pm8775.h          | 146 ++++++++++++++++++
 .../iio/qcom,spmi-adc5-gen3-pmk8550.h         |  54 +++++++
 include/dt-bindings/iio/qcom,spmi-vadc.h      |  79 ++++++++++
 6 files changed, 439 insertions(+)
 create mode 100644 include/dt-bindings/iio/qcom,spmi-adc5-gen3-pm8550.h
 create mode 100644 include/dt-bindings/iio/qcom,spmi-adc5-gen3-pm8550b.h
 create mode 100644 include/dt-bindings/iio/qcom,spmi-adc5-gen3-pm8550vx.h
 create mode 100644 include/dt-bindings/iio/qcom,spmi-adc5-gen3-pm8775.h
 create mode 100644 include/dt-bindings/iio/qcom,spmi-adc5-gen3-pmk8550.h

diff --git a/include/dt-bindings/iio/qcom,spmi-adc5-gen3-pm8550.h b/include/dt-bindings/iio/qcom,spmi-adc5-gen3-pm8550.h
new file mode 100644
index 000000000000..615724cc431a
--- /dev/null
+++ b/include/dt-bindings/iio/qcom,spmi-adc5-gen3-pm8550.h
@@ -0,0 +1,48 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
+/*
+ * Copyright (c) 2024, Qualcomm Innovation Center, Inc. All rights reserved.
+ */
+
+#ifndef _DT_BINDINGS_QCOM_SPMI_VADC_PM8550_H
+#define _DT_BINDINGS_QCOM_SPMI_VADC_PM8550_H
+
+#ifndef PM8550_SID
+#define PM8550_SID		1
+#endif
+
+/* ADC channels for PM8550_ADC for PMIC5 Gen3 */
+#define PM8550_ADC5_GEN3_OFFSET_REF			(PM8550_SID << 8 | 0x00)
+#define PM8550_ADC5_GEN3_1P25VREF			(PM8550_SID << 8 | 0x01)
+#define PM8550_ADC5_GEN3_VREF_VADC			(PM8550_SID << 8 | 0x02)
+#define PM8550_ADC5_GEN3_DIE_TEMP			(PM8550_SID << 8 | 0x03)
+
+#define PM8550_ADC5_GEN3_AMUX_THM1			(PM8550_SID << 8 | 0x04)
+#define PM8550_ADC5_GEN3_AMUX_THM2			(PM8550_SID << 8 | 0x05)
+#define PM8550_ADC5_GEN3_AMUX_THM3			(PM8550_SID << 8 | 0x06)
+#define PM8550_ADC5_GEN3_AMUX_THM4			(PM8550_SID << 8 | 0x07)
+#define PM8550_ADC5_GEN3_AMUX_THM5			(PM8550_SID << 8 | 0x08)
+#define PM8550_ADC5_GEN3_AMUX_THM6_GPIO2		(PM8550_SID << 8 | 0x09)
+#define PM8550_ADC5_GEN3_AMUX1_GPIO3			(PM8550_SID << 8 | 0x0a)
+#define PM8550_ADC5_GEN3_AMUX2_GPIO4			(PM8550_SID << 8 | 0x0b)
+#define PM8550_ADC5_GEN3_AMUX3_GPIO7			(PM8550_SID << 8 | 0x0c)
+#define PM8550_ADC5_GEN3_AMUX4_GPIO12			(PM8550_SID << 8 | 0x0d)
+
+/* 100k pull-up */
+#define PM8550_ADC5_GEN3_AMUX_THM1_100K_PU		(PM8550_SID << 8 | 0x44)
+#define PM8550_ADC5_GEN3_AMUX_THM2_100K_PU		(PM8550_SID << 8 | 0x45)
+#define PM8550_ADC5_GEN3_AMUX_THM3_100K_PU		(PM8550_SID << 8 | 0x46)
+#define PM8550_ADC5_GEN3_AMUX_THM4_100K_PU		(PM8550_SID << 8 | 0x47)
+#define PM8550_ADC5_GEN3_AMUX_THM5_100K_PU		(PM8550_SID << 8 | 0x48)
+#define PM8550_ADC5_GEN3_AMUX_THM6_GPIO2_100K_PU	(PM8550_SID << 8 | 0x49)
+#define PM8550_ADC5_GEN3_AMUX1_GPIO3_100K_PU		(PM8550_SID << 8 | 0x4a)
+#define PM8550_ADC5_GEN3_AMUX2_GPIO4_100K_PU		(PM8550_SID << 8 | 0x4b)
+#define PM8550_ADC5_GEN3_AMUX3_GPIO7_100K_PU		(PM8550_SID << 8 | 0x4c)
+#define PM8550_ADC5_GEN3_AMUX4_GPIO12_100K_PU		(PM8550_SID << 8 | 0x4d)
+
+/* 1/3 Divider */
+#define PM8550_ADC5_GEN3_AMUX3_GPIO7_DIV3		(PM8550_SID << 8 | 0x8c)
+#define PM8550_ADC5_GEN3_AMUX4_GPIO12_DIV3		(PM8550_SID << 8 | 0x8d)
+
+#define PM8550_ADC5_GEN3_VPH_PWR			(PM8550_SID << 8 | 0x8e)
+
+#endif /* _DT_BINDINGS_QCOM_SPMI_VADC_PM8550_H */
diff --git a/include/dt-bindings/iio/qcom,spmi-adc5-gen3-pm8550b.h b/include/dt-bindings/iio/qcom,spmi-adc5-gen3-pm8550b.h
new file mode 100644
index 000000000000..1869eb2a5c6e
--- /dev/null
+++ b/include/dt-bindings/iio/qcom,spmi-adc5-gen3-pm8550b.h
@@ -0,0 +1,97 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
+/*
+ * Copyright (c) 2024, Qualcomm Innovation Center, Inc. All rights reserved.
+ */
+
+#ifndef _DT_BINDINGS_QCOM_SPMI_VADC_PM8550B_H
+#define _DT_BINDINGS_QCOM_SPMI_VADC_PM8550B_H
+
+#ifndef PM8550B_SID
+#define PM8550B_SID		7
+#endif
+
+/* ADC channels for PM8550B_ADC for PMIC5 Gen3 */
+#define PM8550B_ADC5_GEN3_OFFSET_REF			(PM8550B_SID << 8 | 0x00)
+#define PM8550B_ADC5_GEN3_1P25VREF			(PM8550B_SID << 8 | 0x01)
+#define PM8550B_ADC5_GEN3_VREF_VADC			(PM8550B_SID << 8 | 0x02)
+#define PM8550B_ADC5_GEN3_DIE_TEMP			(PM8550B_SID << 8 | 0x03)
+
+#define PM8550B_ADC5_GEN3_AMUX_THM1_BATT_THERM		(PM8550B_SID << 8 | 0x04)
+#define PM8550B_ADC5_GEN3_AMUX_THM2_BATT_ID		(PM8550B_SID << 8 | 0x05)
+#define PM8550B_ADC5_GEN3_AMUX_THM3_SMB_TEMP_V		(PM8550B_SID << 8 | 0x06)
+#define PM8550B_ADC5_GEN3_AMUX_THM4_USB_THERM		(PM8550B_SID << 8 | 0x07)
+#define PM8550B_ADC5_GEN3_AMUX_THM5_OPTION		(PM8550B_SID << 8 | 0x08)
+#define PM8550B_ADC5_GEN3_AMUX_THM6_GPIO10		(PM8550B_SID << 8 | 0x09)
+#define PM8550B_ADC5_GEN3_AMUX1_GPIO1			(PM8550B_SID << 8 | 0x0a)
+#define PM8550B_ADC5_GEN3_AMUX2_GPIO5			(PM8550B_SID << 8 | 0x0b)
+#define PM8550B_ADC5_GEN3_AMUX3_GPIO6			(PM8550B_SID << 8 | 0x0c)
+#define PM8550B_ADC5_GEN3_AMUX4_GPIO12			(PM8550B_SID << 8 | 0x0d)
+
+#define PM8550B_ADC5_GEN3_CHG_TEMP			(PM8550B_SID << 8 | 0x10)
+#define PM8550B_ADC5_GEN3_USB_SNS_V_16			(PM8550B_SID << 8 | 0x11)
+#define PM8550B_ADC5_GEN3_VIN_DIV16_MUX			(PM8550B_SID << 8 | 0x12)
+#define PM8550B_ADC5_GEN3_USBC_MUX			(PM8550B_SID << 8 | 0x13)
+#define PM8550B_ADC5_GEN3_VREF_BAT_THERM		(PM8550B_SID << 8 | 0x15)
+#define PM8550B_ADC5_GEN3_IIN_FB			(PM8550B_SID << 8 | 0x17)
+#define PM8550B_ADC5_GEN3_TEMP_ALARM_LITE		(PM8550B_SID << 8 | 0x18)
+#define PM8550B_ADC5_GEN3_SMB_IIN			(PM8550B_SID << 8 | 0x19)
+#define PM8550B_ADC5_GEN3_VREF_BAT2_THERM		(PM8550B_SID << 8 | 0x1a)
+#define PM8550B_ADC5_GEN3_SMB_ICHG			(PM8550B_SID << 8 | 0x1b)
+#define PM8550B_ADC5_GEN3_SMB_TEMP_I			(PM8550B_SID << 8 | 0x1e)
+#define PM8550B_ADC5_GEN3_CHG_TEMP_I			(PM8550B_SID << 8 | 0x1f)
+#define PM8550B_ADC5_GEN3_ICHG_FB			(PM8550B_SID << 8 | 0xa1)
+
+/* 30k pull-up */
+#define PM8550B_ADC5_GEN3_AMUX_THM1_BATT_THERM_30K_PU	(PM8550B_SID << 8 | 0x24)
+#define PM8550B_ADC5_GEN3_AMUX_THM2_BATT_ID_30K_PU	(PM8550B_SID << 8 | 0x25)
+#define PM8550B_ADC5_GEN3_AMUX_THM3_SMB_TEMP_V_30K_PU	(PM8550B_SID << 8 | 0x26)
+#define PM8550B_ADC5_GEN3_AMUX_THM4_USB_THERM_30K_PU	(PM8550B_SID << 8 | 0x27)
+#define PM8550B_ADC5_GEN3_AMUX_THM5_OPTION_30K_PU	(PM8550B_SID << 8 | 0x28)
+#define PM8550B_ADC5_GEN3_AMUX_THM6_GPIO10_30K_PU	(PM8550B_SID << 8 | 0x29)
+#define PM8550B_ADC5_GEN3_AMUX1_GPIO1_30K_PU		(PM8550B_SID << 8 | 0x2a)
+#define PM8550B_ADC5_GEN3_AMUX2_GPIO5_30K_PU		(PM8550B_SID << 8 | 0x2b)
+#define PM8550B_ADC5_GEN3_AMUX3_GPIO6_30K_PU		(PM8550B_SID << 8 | 0x2c)
+#define PM8550B_ADC5_GEN3_AMUX4_GPIO12_30K_PU		(PM8550B_SID << 8 | 0x2d)
+
+#define PM8550B_ADC5_GEN3_USBC_MUX_30K_PU		(PM8550B_SID << 8 | 0x33)
+
+/* 100k pull-up */
+#define PM8550B_ADC5_GEN3_AMUX_THM1_BATT_THERM_100K_PU	(PM8550B_SID << 8 | 0x44)
+#define PM8550B_ADC5_GEN3_AMUX_THM2_BATT_ID_100K_PU	(PM8550B_SID << 8 | 0x45)
+#define PM8550B_ADC5_GEN3_AMUX_THM3_SMB_TEMP_V_100K_PU	(PM8550B_SID << 8 | 0x46)
+#define PM8550B_ADC5_GEN3_AMUX_THM4_USB_THERM_100K_PU	(PM8550B_SID << 8 | 0x47)
+#define PM8550B_ADC5_GEN3_AMUX_THM5_OPTION_100K_PU	(PM8550B_SID << 8 | 0x48)
+#define PM8550B_ADC5_GEN3_AMUX_THM6_GPIO10_100K_PU	(PM8550B_SID << 8 | 0x49)
+#define PM8550B_ADC5_GEN3_AMUX1_GPIO1_100K_PU		(PM8550B_SID << 8 | 0x4a)
+#define PM8550B_ADC5_GEN3_AMUX2_GPIO5_100K_PU		(PM8550B_SID << 8 | 0x4b)
+#define PM8550B_ADC5_GEN3_AMUX3_GPIO6_100K_PU		(PM8550B_SID << 8 | 0x4c)
+#define PM8550B_ADC5_GEN3_AMUX4_GPIO12_100K_PU		(PM8550B_SID << 8 | 0x4d)
+
+#define PM8550B_ADC5_GEN3_USBC_MUX_100K_PU		(PM8550B_SID << 8 | 0x53)
+
+/* 400k pull-up */
+#define PM8550B_ADC5_GEN3_AMUX_THM1_BATT_THERM_400K_PU	(PM8550B_SID << 8 | 0x64)
+#define PM8550B_ADC5_GEN3_AMUX_THM2_BATT_ID_400K_PU	(PM8550B_SID << 8 | 0x65)
+#define PM8550B_ADC5_GEN3_AMUX_THM3_SMB_TEMP_V_400K_PU	(PM8550B_SID << 8 | 0x66)
+#define PM8550B_ADC5_GEN3_AMUX_THM4_USB_THERM_400K_PU	(PM8550B_SID << 8 | 0x67)
+#define PM8550B_ADC5_GEN3_AMUX_THM5_OPTION_400K_PU	(PM8550B_SID << 8 | 0x68)
+#define PM8550B_ADC5_GEN3_AMUX_THM6_GPIO10_400K_PU	(PM8550B_SID << 8 | 0x69)
+#define PM8550B_ADC5_GEN3_AMUX1_GPIO1_400K_PU		(PM8550B_SID << 8 | 0x6a)
+#define PM8550B_ADC5_GEN3_AMUX2_GPIO5_400K_PU		(PM8550B_SID << 8 | 0x6b)
+#define PM8550B_ADC5_GEN3_AMUX3_GPIO6_400K_PU		(PM8550B_SID << 8 | 0x6c)
+#define PM8550B_ADC5_GEN3_AMUX4_GPIO12_400K_PU		(PM8550B_SID << 8 | 0x6d)
+
+#define PM8550B_ADC5_GEN3_USBC_MUX_400K_PU		(PM8550B_SID << 8 | 0x73)
+
+/* 1/3 Divider */
+#define PM8550B_ADC5_GEN3_AMUX1_GPIO1_DIV3		(PM8550B_SID << 8 | 0x8a)
+#define PM8550B_ADC5_GEN3_AMUX2_GPIO5_DIV3		(PM8550B_SID << 8 | 0x8b)
+#define PM8550B_ADC5_GEN3_AMUX3_GPIO6_DIV3		(PM8550B_SID << 8 | 0x8c)
+
+#define PM8550B_ADC5_GEN3_VPH_PWR			(PM8550B_SID << 8 | 0x8e)
+#define PM8550B_ADC5_GEN3_VBAT_SNS_QBG			(PM8550B_SID << 8 | 0x8f)
+#define PM8550B_ADC5_GEN3_VBAT_SNS_CHGR			(PM8550B_SID << 8 | 0x94)
+#define PM8550B_ADC5_GEN3_VBAT_2S_MID_QBG		(PM8550B_SID << 8 | 0x96)
+#define PM8550B_ADC5_GEN3_VBAT_2S_MID_CHGR		(PM8550B_SID << 8 | 0x9d)
+
+#endif /* _DT_BINDINGS_QCOM_SPMI_VADC_PM8550B_H */
diff --git a/include/dt-bindings/iio/qcom,spmi-adc5-gen3-pm8550vx.h b/include/dt-bindings/iio/qcom,spmi-adc5-gen3-pm8550vx.h
new file mode 100644
index 000000000000..c2ed7fa1adb1
--- /dev/null
+++ b/include/dt-bindings/iio/qcom,spmi-adc5-gen3-pm8550vx.h
@@ -0,0 +1,15 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
+/*
+ * Copyright (c) 2024, Qualcomm Innovation Center, Inc. All rights reserved.
+ */
+
+#ifndef _DT_BINDINGS_QCOM_SPMI_VADC_PM8550VX_H
+#define _DT_BINDINGS_QCOM_SPMI_VADC_PM8550VX_H
+
+/* ADC channels for PM8550VX_ADC for PMIC5 Gen3 */
+#define PM8550VX_ADC5_GEN3_OFFSET_REF(sid)		((sid) << 8 | 0x00)
+#define PM8550VX_ADC5_GEN3_1P25VREF(sid)		((sid) << 8 | 0x01)
+#define PM8550VX_ADC5_GEN3_VREF_VADC(sid)		((sid) << 8 | 0x02)
+#define PM8550VX_ADC5_GEN3_DIE_TEMP(sid)		((sid) << 8 | 0x03)
+
+#endif /* _DT_BINDINGS_QCOM_SPMI_VADC_PM8550VX_H */
diff --git a/include/dt-bindings/iio/qcom,spmi-adc5-gen3-pm8775.h b/include/dt-bindings/iio/qcom,spmi-adc5-gen3-pm8775.h
new file mode 100644
index 000000000000..784724944ac4
--- /dev/null
+++ b/include/dt-bindings/iio/qcom,spmi-adc5-gen3-pm8775.h
@@ -0,0 +1,146 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
+/*
+ * Copyright (c) 2024, Qualcomm Innovation Center, Inc. All rights reserved.
+ */
+
+#ifndef _DT_BINDINGS_QCOM_SPMI_VADC_PM8775_H
+#define _DT_BINDINGS_QCOM_SPMI_VADC_PM8775_H
+
+#ifndef PM8775_1_SID
+#define PM8775_1_SID		0
+#endif
+#ifndef PM8775_2_SID
+#define PM8775_2_SID		2
+#endif
+#ifndef PM8775_3_SID
+#define PM8775_3_SID		4
+#endif
+#ifndef PM8775_4_SID
+#define PM8775_4_SID		6
+#endif
+
+/* ADC channels for PM8775_1_ADC for PMIC5 Gen3 */
+#define PM8775_1_ADC5_GEN3_OFFSET_REF			(PM8775_1_SID << 8 | 0x0)
+#define PM8775_1_ADC5_GEN3_1P25VREF			(PM8775_1_SID << 8 | 0x01)
+#define PM8775_1_ADC5_GEN3_VREF_VADC			(PM8775_1_SID << 8 | 0x02)
+#define PM8775_1_ADC5_GEN3_DIE_TEMP			(PM8775_1_SID << 8 | 0x03)
+
+#define PM8775_1_ADC5_GEN3_AMUX1_THM			(PM8775_1_SID << 8 | 0x04)
+#define PM8775_1_ADC5_GEN3_AMUX2_THM			(PM8775_1_SID << 8 | 0x05)
+#define PM8775_1_ADC5_GEN3_AMUX3_THM			(PM8775_1_SID << 8 | 0x06)
+#define PM8775_1_ADC5_GEN3_AMUX4_THM			(PM8775_1_SID << 8 | 0x07)
+#define PM8775_1_ADC5_GEN3_AMUX5_THM			(PM8775_1_SID << 8 | 0x08)
+#define PM8775_1_ADC5_GEN3_AMUX6_THM			(PM8775_1_SID << 8 | 0x09)
+#define PM8775_1_ADC5_GEN3_AMUX1_GPIO9			(PM8775_1_SID << 8 | 0x0a)
+#define PM8775_1_ADC5_GEN3_AMUX2_GPIO10			(PM8775_1_SID << 8 | 0x0b)
+#define PM8775_1_ADC5_GEN3_AMUX3_GPIO11			(PM8775_1_SID << 8 | 0x0c)
+#define PM8775_1_ADC5_GEN3_AMUX4_GPIO12			(PM8775_1_SID << 8 | 0x0d)
+
+/* 100k pull-up2 */
+#define PM8775_1_ADC5_GEN3_AMUX1_THM_100K_PU		(PM8775_1_SID << 8 | 0x44)
+#define PM8775_1_ADC5_GEN3_AMUX2_THM_100K_PU		(PM8775_1_SID << 8 | 0x45)
+#define PM8775_1_ADC5_GEN3_AMUX3_THM_100K_PU		(PM8775_1_SID << 8 | 0x46)
+#define PM8775_1_ADC5_GEN3_AMUX4_THM_100K_PU		(PM8775_1_SID << 8 | 0x47)
+#define PM8775_1_ADC5_GEN3_AMUX5_THM_100K_PU		(PM8775_1_SID << 8 | 0x48)
+#define PM8775_1_ADC5_GEN3_AMUX6_THM_100K_PU		(PM8775_1_SID << 8 | 0x49)
+#define PM8775_1_ADC5_GEN3_AMUX1_GPIO9_100K_PU		(PM8775_1_SID << 8 | 0x4a)
+#define PM8775_1_ADC5_GEN3_AMUX2_GPIO10_100K_PU		(PM8775_1_SID << 8 | 0x4b)
+#define PM8775_1_ADC5_GEN3_AMUX3_GPIO11_100K_PU		(PM8775_1_SID << 8 | 0x4c)
+#define PM8775_1_ADC5_GEN3_AMUX4_GPIO12_100K_PU		(PM8775_1_SID << 8 | 0x4d)
+
+#define PM8775_1_ADC5_GEN3_VPH_PWR			(PM8775_1_SID << 8 | 0x8e)
+
+/* ADC channels for PM8775_2_ADC for PMIC5 Gen3 */
+#define PM8775_2_ADC5_GEN3_OFFSET_REF			(PM8775_2_SID << 8 | 0x0)
+#define PM8775_2_ADC5_GEN3_1P25VREF			(PM8775_2_SID << 8 | 0x01)
+#define PM8775_2_ADC5_GEN3_VREF_VADC			(PM8775_2_SID << 8 | 0x02)
+#define PM8775_2_ADC5_GEN3_DIE_TEMP			(PM8775_2_SID << 8 | 0x03)
+
+#define PM8775_2_ADC5_GEN3_AMUX1_THM			(PM8775_2_SID << 8 | 0x04)
+#define PM8775_2_ADC5_GEN3_AMUX2_THM			(PM8775_2_SID << 8 | 0x05)
+#define PM8775_2_ADC5_GEN3_AMUX3_THM			(PM8775_2_SID << 8 | 0x06)
+#define PM8775_2_ADC5_GEN3_AMUX4_THM			(PM8775_2_SID << 8 | 0x07)
+#define PM8775_2_ADC5_GEN3_AMUX5_THM			(PM8775_2_SID << 8 | 0x08)
+#define PM8775_2_ADC5_GEN3_AMUX6_THM			(PM8775_2_SID << 8 | 0x09)
+#define PM8775_2_ADC5_GEN3_AMUX1_GPIO9			(PM8775_2_SID << 8 | 0x0a)
+#define PM8775_2_ADC5_GEN3_AMUX2_GPIO10			(PM8775_2_SID << 8 | 0x0b)
+#define PM8775_2_ADC5_GEN3_AMUX3_GPIO11			(PM8775_2_SID << 8 | 0x0c)
+#define PM8775_2_ADC5_GEN3_AMUX4_GPIO12			(PM8775_2_SID << 8 | 0x0d)
+
+/* 100k pull-up2 */
+#define PM8775_2_ADC5_GEN3_AMUX1_THM_100K_PU		(PM8775_2_SID << 8 | 0x44)
+#define PM8775_2_ADC5_GEN3_AMUX2_THM_100K_PU		(PM8775_2_SID << 8 | 0x45)
+#define PM8775_2_ADC5_GEN3_AMUX3_THM_100K_PU		(PM8775_2_SID << 8 | 0x46)
+#define PM8775_2_ADC5_GEN3_AMUX4_THM_100K_PU		(PM8775_2_SID << 8 | 0x47)
+#define PM8775_2_ADC5_GEN3_AMUX5_THM_100K_PU		(PM8775_2_SID << 8 | 0x48)
+#define PM8775_2_ADC5_GEN3_AMUX6_THM_100K_PU		(PM8775_2_SID << 8 | 0x49)
+#define PM8775_2_ADC5_GEN3_AMUX1_GPIO9_100K_PU		(PM8775_2_SID << 8 | 0x4a)
+#define PM8775_2_ADC5_GEN3_AMUX2_GPIO10_100K_PU		(PM8775_2_SID << 8 | 0x4b)
+#define PM8775_2_ADC5_GEN3_AMUX3_GPIO11_100K_PU		(PM8775_2_SID << 8 | 0x4c)
+#define PM8775_2_ADC5_GEN3_AMUX4_GPIO12_100K_PU		(PM8775_2_SID << 8 | 0x4d)
+
+#define PM8775_2_ADC5_GEN3_VPH_PWR			(PM8775_2_SID << 8 | 0x8e)
+
+/* ADC channels for PM8775_3_ADC for PMIC5 Gen3 */
+#define PM8775_3_ADC5_GEN3_OFFSET_REF			(PM8775_3_SID << 8 | 0x0)
+#define PM8775_3_ADC5_GEN3_1P25VREF			(PM8775_3_SID << 8 | 0x01)
+#define PM8775_3_ADC5_GEN3_VREF_VADC			(PM8775_3_SID << 8 | 0x02)
+#define PM8775_3_ADC5_GEN3_DIE_TEMP			(PM8775_3_SID << 8 | 0x03)
+
+#define PM8775_3_ADC5_GEN3_AMUX1_THM			(PM8775_3_SID << 8 | 0x04)
+#define PM8775_3_ADC5_GEN3_AMUX2_THM			(PM8775_3_SID << 8 | 0x05)
+#define PM8775_3_ADC5_GEN3_AMUX3_THM			(PM8775_3_SID << 8 | 0x06)
+#define PM8775_3_ADC5_GEN3_AMUX4_THM			(PM8775_3_SID << 8 | 0x07)
+#define PM8775_3_ADC5_GEN3_AMUX5_THM			(PM8775_3_SID << 8 | 0x08)
+#define PM8775_3_ADC5_GEN3_AMUX6_THM			(PM8775_3_SID << 8 | 0x09)
+#define PM8775_3_ADC5_GEN3_AMUX1_GPIO9			(PM8775_3_SID << 8 | 0x0a)
+#define PM8775_3_ADC5_GEN3_AMUX2_GPIO10			(PM8775_3_SID << 8 | 0x0b)
+#define PM8775_3_ADC5_GEN3_AMUX3_GPIO11			(PM8775_3_SID << 8 | 0x0c)
+#define PM8775_3_ADC5_GEN3_AMUX4_GPIO12			(PM8775_3_SID << 8 | 0x0d)
+
+/* 100k pull-up2 */
+#define PM8775_3_ADC5_GEN3_AMUX1_THM_100K_PU		(PM8775_3_SID << 8 | 0x44)
+#define PM8775_3_ADC5_GEN3_AMUX2_THM_100K_PU		(PM8775_3_SID << 8 | 0x45)
+#define PM8775_3_ADC5_GEN3_AMUX3_THM_100K_PU		(PM8775_3_SID << 8 | 0x46)
+#define PM8775_3_ADC5_GEN3_AMUX4_THM_100K_PU		(PM8775_3_SID << 8 | 0x47)
+#define PM8775_3_ADC5_GEN3_AMUX5_THM_100K_PU		(PM8775_3_SID << 8 | 0x48)
+#define PM8775_3_ADC5_GEN3_AMUX6_THM_100K_PU		(PM8775_3_SID << 8 | 0x49)
+#define PM8775_3_ADC5_GEN3_AMUX1_GPIO9_100K_PU		(PM8775_3_SID << 8 | 0x4a)
+#define PM8775_3_ADC5_GEN3_AMUX2_GPIO10_100K_PU		(PM8775_3_SID << 8 | 0x4b)
+#define PM8775_3_ADC5_GEN3_AMUX3_GPIO11_100K_PU		(PM8775_3_SID << 8 | 0x4c)
+#define PM8775_3_ADC5_GEN3_AMUX4_GPIO12_100K_PU		(PM8775_3_SID << 8 | 0x4d)
+
+#define PM8775_3_ADC5_GEN3_VPH_PWR			(PM8775_3_SID << 8 | 0x8e)
+
+/* ADC channels for PM8775_4_ADC for PMIC5 Gen3 */
+#define PM8775_4_ADC5_GEN3_OFFSET_REF			(PM8775_4_SID << 8 | 0x0)
+#define PM8775_4_ADC5_GEN3_1P25VREF			(PM8775_4_SID << 8 | 0x01)
+#define PM8775_4_ADC5_GEN3_VREF_VADC			(PM8775_4_SID << 8 | 0x02)
+#define PM8775_4_ADC5_GEN3_DIE_TEMP			(PM8775_4_SID << 8 | 0x03)
+
+#define PM8775_4_ADC5_GEN3_AMUX1_THM			(PM8775_4_SID << 8 | 0x04)
+#define PM8775_4_ADC5_GEN3_AMUX2_THM			(PM8775_4_SID << 8 | 0x05)
+#define PM8775_4_ADC5_GEN3_AMUX3_THM			(PM8775_4_SID << 8 | 0x06)
+#define PM8775_4_ADC5_GEN3_AMUX4_THM			(PM8775_4_SID << 8 | 0x07)
+#define PM8775_4_ADC5_GEN3_AMUX5_THM			(PM8775_4_SID << 8 | 0x08)
+#define PM8775_4_ADC5_GEN3_AMUX6_THM			(PM8775_4_SID << 8 | 0x09)
+#define PM8775_4_ADC5_GEN3_AMUX1_GPIO9			(PM8775_4_SID << 8 | 0x0a)
+#define PM8775_4_ADC5_GEN3_AMUX2_GPIO10			(PM8775_4_SID << 8 | 0x0b)
+#define PM8775_4_ADC5_GEN3_AMUX3_GPIO11			(PM8775_4_SID << 8 | 0x0c)
+#define PM8775_4_ADC5_GEN3_AMUX4_GPIO12			(PM8775_4_SID << 8 | 0x0d)
+
+/* 100k pull-up2 */
+#define PM8775_4_ADC5_GEN3_AMUX1_THM_100K_PU		(PM8775_4_SID << 8 | 0x44)
+#define PM8775_4_ADC5_GEN3_AMUX2_THM_100K_PU		(PM8775_4_SID << 8 | 0x45)
+#define PM8775_4_ADC5_GEN3_AMUX3_THM_100K_PU		(PM8775_4_SID << 8 | 0x46)
+#define PM8775_4_ADC5_GEN3_AMUX4_THM_100K_PU		(PM8775_4_SID << 8 | 0x47)
+#define PM8775_4_ADC5_GEN3_AMUX5_THM_100K_PU		(PM8775_4_SID << 8 | 0x48)
+#define PM8775_4_ADC5_GEN3_AMUX6_THM_100K_PU		(PM8775_4_SID << 8 | 0x49)
+#define PM8775_4_ADC5_GEN3_AMUX1_GPIO9_100K_PU		(PM8775_4_SID << 8 | 0x4a)
+#define PM8775_4_ADC5_GEN3_AMUX2_GPIO10_100K_PU		(PM8775_4_SID << 8 | 0x4b)
+#define PM8775_4_ADC5_GEN3_AMUX3_GPIO11_100K_PU		(PM8775_4_SID << 8 | 0x4c)
+#define PM8775_4_ADC5_GEN3_AMUX4_GPIO12_100K_PU		(PM8775_4_SID << 8 | 0x4d)
+
+#define PM8775_4_ADC5_GEN3_VPH_PWR			(PM8775_4_SID << 8 | 0x8e)
+
+#endif /* _DT_BINDINGS_QCOM_SPMI_VADC_PM8775_H */
diff --git a/include/dt-bindings/iio/qcom,spmi-adc5-gen3-pmk8550.h b/include/dt-bindings/iio/qcom,spmi-adc5-gen3-pmk8550.h
new file mode 100644
index 000000000000..1ebb1ba76bf2
--- /dev/null
+++ b/include/dt-bindings/iio/qcom,spmi-adc5-gen3-pmk8550.h
@@ -0,0 +1,54 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
+/*
+ * Copyright (c) 2024, Qualcomm Innovation Center, Inc. All rights reserved.
+ */
+
+#ifndef _DT_BINDINGS_QCOM_SPMI_VADC_PMK8550_H
+#define _DT_BINDINGS_QCOM_SPMI_VADC_PMK8550_H
+
+#ifndef PMK8550_SID
+#define PMK8550_SID		0
+#endif
+
+/* ADC channels for PMK8550_ADC for PMIC5 Gen3 */
+#define PMK8550_ADC5_GEN3_OFFSET_REF			(PMK8550_SID << 8 | 0x00)
+#define PMK8550_ADC5_GEN3_1P25VREF			(PMK8550_SID << 8 | 0x01)
+#define PMK8550_ADC5_GEN3_VREF_VADC			(PMK8550_SID << 8 | 0x02)
+#define PMK8550_ADC5_GEN3_DIE_TEMP			(PMK8550_SID << 8 | 0x03)
+
+#define PMK8550_ADC5_GEN3_AMUX_THM1_XO_THERM		(PMK8550_SID << 8 | 0x04)
+#define PMK8550_ADC5_GEN3_AMUX_THM2_GPIO1		(PMK8550_SID << 8 | 0x05)
+#define PMK8550_ADC5_GEN3_AMUX_THM3_GPIO2		(PMK8550_SID << 8 | 0x06)
+#define PMK8550_ADC5_GEN3_AMUX_THM4_GPIO3		(PMK8550_SID << 8 | 0x07)
+#define PMK8550_ADC5_GEN3_AMUX_THM5_GPIO4		(PMK8550_SID << 8 | 0x08)
+#define PMK8550_ADC5_GEN3_AMUX_THM6_GPIO5		(PMK8550_SID << 8 | 0x09)
+#define PMK8550_ADC5_GEN3_AMUX1_GPIO6			(PMK8550_SID << 8 | 0x0a)
+
+/* 30k pull-up */
+#define PMK8550_ADC5_GEN3_AMUX_THM1_XO_THERM_30K_PU	(PMK8550_SID << 8 | 0x24)
+#define PMK8550_ADC5_GEN3_AMUX_THM2_GPIO1_30K_PU	(PMK8550_SID << 8 | 0x25)
+#define PMK8550_ADC5_GEN3_AMUX_THM3_GPIO2_30K_PU	(PMK8550_SID << 8 | 0x26)
+#define PMK8550_ADC5_GEN3_AMUX_THM4_GPIO3_30K_PU	(PMK8550_SID << 8 | 0x27)
+#define PMK8550_ADC5_GEN3_AMUX_THM5_GPIO4_30K_PU	(PMK8550_SID << 8 | 0x28)
+#define PMK8550_ADC5_GEN3_AMUX_THM6_GPIO5_30K_PU	(PMK8550_SID << 8 | 0x29)
+#define PMK8550_ADC5_GEN3_AMUX1_GPIO6_30K_PU		(PMK8550_SID << 8 | 0x2a)
+
+/* 100k pull-up */
+#define PMK8550_ADC5_GEN3_AMUX_THM1_XO_THERM_100K_PU	(PMK8550_SID << 8 | 0x44)
+#define PMK8550_ADC5_GEN3_AMUX_THM2_GPIO1_100K_PU	(PMK8550_SID << 8 | 0x45)
+#define PMK8550_ADC5_GEN3_AMUX_THM3_GPIO2_100K_PU	(PMK8550_SID << 8 | 0x46)
+#define PMK8550_ADC5_GEN3_AMUX_THM4_GPIO3_100K_PU	(PMK8550_SID << 8 | 0x47)
+#define PMK8550_ADC5_GEN3_AMUX_THM5_GPIO4_100K_PU	(PMK8550_SID << 8 | 0x48)
+#define PMK8550_ADC5_GEN3_AMUX_THM6_GPIO5_100K_PU	(PMK8550_SID << 8 | 0x49)
+#define PMK8550_ADC5_GEN3_AMUX1_GPIO6_100K_PU		(PMK8550_SID << 8 | 0x4a)
+
+/* 400k pull-up */
+#define PMK8550_ADC5_GEN3_AMUX_THM1_XO_THERM_400K_PU	(PMK8550_SID << 8 | 0x64)
+#define PMK8550_ADC5_GEN3_AMUX_THM2_GPIO1_400K_PU	(PMK8550_SID << 8 | 0x65)
+#define PMK8550_ADC5_GEN3_AMUX_THM3_GPIO2_400K_PU	(PMK8550_SID << 8 | 0x66)
+#define PMK8550_ADC5_GEN3_AMUX_THM4_GPIO3_400K_PU	(PMK8550_SID << 8 | 0x67)
+#define PMK8550_ADC5_GEN3_AMUX_THM5_GPIO4_400K_PU	(PMK8550_SID << 8 | 0x68)
+#define PMK8550_ADC5_GEN3_AMUX_THM6_GPIO5_400K_PU	(PMK8550_SID << 8 | 0x69)
+#define PMK8550_ADC5_GEN3_AMUX1_GPIO6_400K_PU		(PMK8550_SID << 8 | 0x6a)
+
+#endif /* _DT_BINDINGS_QCOM_SPMI_VADC_PMK8550_H */
diff --git a/include/dt-bindings/iio/qcom,spmi-vadc.h b/include/dt-bindings/iio/qcom,spmi-vadc.h
index 08adfe25964c..275d5bf1c45f 100644
--- a/include/dt-bindings/iio/qcom,spmi-vadc.h
+++ b/include/dt-bindings/iio/qcom,spmi-vadc.h
@@ -1,6 +1,7 @@
 /* SPDX-License-Identifier: GPL-2.0 */
 /*
  * Copyright (c) 2012-2014,2018,2020 The Linux Foundation. All rights reserved.
+ * Copyright (c) 2024, Qualcomm Innovation Center, Inc. All rights reserved.
  */
 
 #ifndef _DT_BINDINGS_QCOM_SPMI_VADC_H
@@ -297,4 +298,82 @@
 #define ADC7_SBUx				0x94
 #define ADC7_VBAT_2S_MID			0x96
 
+/* ADC channels for PMIC5 Gen3 */
+
+#define ADC5_GEN3_OFFSET_REF			0x00
+#define ADC5_GEN3_1P25VREF			0x01
+#define ADC5_GEN3_VREF_VADC			0x02
+#define ADC5_GEN3_DIE_TEMP			0x03
+
+#define ADC5_GEN3_AMUX1_THM			0x04
+#define ADC5_GEN3_AMUX2_THM			0x05
+#define ADC5_GEN3_AMUX3_THM			0x06
+#define ADC5_GEN3_AMUX4_THM			0x07
+#define ADC5_GEN3_AMUX5_THM			0x08
+#define ADC5_GEN3_AMUX6_THM			0x09
+#define ADC5_GEN3_AMUX1_GPIO			0x0a
+#define ADC5_GEN3_AMUX2_GPIO			0x0b
+#define ADC5_GEN3_AMUX3_GPIO			0x0c
+#define ADC5_GEN3_AMUX4_GPIO			0x0d
+
+#define ADC5_GEN3_CHG_TEMP			0x10
+#define ADC5_GEN3_USB_SNS_V_16			0x11
+#define ADC5_GEN3_VIN_DIV16_MUX			0x12
+#define ADC5_GEN3_VREF_BAT_THERM		0x15
+#define ADC5_GEN3_IIN_FB			0x17
+#define ADC5_GEN3_TEMP_ALARM_LITE		0x18
+#define ADC5_GEN3_IIN_SMB			0x19
+#define ADC5_GEN3_ICHG_SMB			0x1b
+#define ADC5_GEN3_ICHG_FB			0xa1
+
+/* 30k pull-up1 */
+#define ADC5_GEN3_AMUX1_THM_30K_PU		0x24
+#define ADC5_GEN3_AMUX2_THM_30K_PU		0x25
+#define ADC5_GEN3_AMUX3_THM_30K_PU		0x26
+#define ADC5_GEN3_AMUX4_THM_30K_PU		0x27
+#define ADC5_GEN3_AMUX5_THM_30K_PU		0x28
+#define ADC5_GEN3_AMUX6_THM_30K_PU		0x29
+#define ADC5_GEN3_AMUX1_GPIO_30K_PU		0x2a
+#define ADC5_GEN3_AMUX2_GPIO_30K_PU		0x2b
+#define ADC5_GEN3_AMUX3_GPIO_30K_PU		0x2c
+#define ADC5_GEN3_AMUX4_GPIO_30K_PU		0x2d
+
+/* 100k pull-up2 */
+#define ADC5_GEN3_AMUX1_THM_100K_PU		0x44
+#define ADC5_GEN3_AMUX2_THM_100K_PU		0x45
+#define ADC5_GEN3_AMUX3_THM_100K_PU		0x46
+#define ADC5_GEN3_AMUX4_THM_100K_PU		0x47
+#define ADC5_GEN3_AMUX5_THM_100K_PU		0x48
+#define ADC5_GEN3_AMUX6_THM_100K_PU		0x49
+#define ADC5_GEN3_AMUX1_GPIO_100K_PU		0x4a
+#define ADC5_GEN3_AMUX2_GPIO_100K_PU		0x4b
+#define ADC5_GEN3_AMUX3_GPIO_100K_PU		0x4c
+#define ADC5_GEN3_AMUX4_GPIO_100K_PU		0x4d
+
+/* 400k pull-up3 */
+#define ADC5_GEN3_AMUX1_THM_400K_PU		0x64
+#define ADC5_GEN3_AMUX2_THM_400K_PU		0x65
+#define ADC5_GEN3_AMUX3_THM_400K_PU		0x66
+#define ADC5_GEN3_AMUX4_THM_400K_PU		0x67
+#define ADC5_GEN3_AMUX5_THM_400K_PU		0x68
+#define ADC5_GEN3_AMUX6_THM_400K_PU		0x69
+#define ADC5_GEN3_AMUX1_GPIO_400K_PU		0x6a
+#define ADC5_GEN3_AMUX2_GPIO_400K_PU		0x6b
+#define ADC5_GEN3_AMUX3_GPIO_400K_PU		0x6c
+#define ADC5_GEN3_AMUX4_GPIO_400K_PU		0x6d
+
+/* 1/3 Divider */
+#define ADC5_GEN3_AMUX1_GPIO_DIV3		0x8a
+#define ADC5_GEN3_AMUX2_GPIO_DIV3		0x8b
+#define ADC5_GEN3_AMUX3_GPIO_DIV3		0x8c
+
+#define ADC5_GEN3_VPH_PWR			0x8e
+#define ADC5_GEN3_VBAT_SNS_QBG			0x8f
+
+#define ADC5_GEN3_VBAT_SNS_CHGR			0x94
+#define ADC5_GEN3_VBAT_2S_MID_QBG		0x96
+#define ADC5_GEN3_VBAT_2S_MID_CHGR		0x9d
+
+#define ADC5_OFFSET_EXT2			0xf8
+
 #endif /* _DT_BINDINGS_QCOM_SPMI_VADC_H */
-- 
2.34.1

